tablo/models.py

```diff
@@ -432,25 +451,32 @@ def _validate_where_clause(self, where):
         parsed = self._parse_where_clause(where)
 
         if parsed is None:
-            return True
+            return
         elif parsed[1]:
-            return False  # More than one statement means SQL injection.
-        else:
-            return self._validate_fields(parsed[0])
+            raise SQLInjectionError('Invalid where clause')
```

Comment:
I meant something different: `InvalidFieldsError` means fields specified in the SQL are either incorrect or not allowed, and it implies there is something the user can do to fix an unintentionally broken query. It also implies that the error message raised with an `InvalidFieldsError` should include useful information about which fields were invalid.

Originally I planned to handle `SQLInjectionError` differently in the `except` block. But because I  made it extend `django.db.utils.DatabaseError`, and because it seemed a generic error was the best way to respond to an attempt at SQL injection, it turned out there was nothing different to do.

But there is still a reason for having a separate exception type in this case: other projects that eventually leverage tablo will need at least some specificity when it comes to errors returned. This is similar to the way the `requests` library and `django.db.utils` breaks out specific errors according to type.

However, I can see an argument to rename `SQLInjectionError` to `InvalidSQLError`. It would mean that instead of representing a very specific issue with the SQL, that all SQL errors should inherit from it (similar to `HTTPError` and its siblings inheriting from `RequestException`). But at this point I'm not sure it's practical to deepen the tablo error inheritance hierarchy, unless you feel strongly about it.